hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/metrics2/util/SampleQuantiles.java

```diff
@@ -243,7 +245,12 @@ synchronized public Map<Quantile, Long> snapshot() {
     
     Map<Quantile, Long> values = new TreeMap<Quantile, Long>();
     for (int i = 0; i < quantiles.length; i++) {
-      values.put(quantiles[i], query(quantiles[i].quantile));
+      /* eg : effectiveQuantile for 0.99 with inverseQuantiles will be 0.01. 
+      For inverse quantiles higher numeric value is better and hence we want 
+      to query from the opposite end of the sorted sample
+       */
+      double effectiveQuantile = inverseQuantiles ? 1 - quantiles[i].quantile : quantiles[i].quantile;
```

Comment:
Shouldn't this come after the next line?

Reply:
This is to pass the effectiveQuantile to the query function hence calculated before.